satellite/console/service.go

```diff
@@ -168,6 +168,142 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 	return u, nil
 }
 
+// AddNewPaymentMethod adds new payment method for project
+func (s *Service) AddNewPaymentMethod(ctx context.Context, paymentMethodToken string, isDefault bool, projectID uuid.UUID) (payment *ProjectPayment, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	authorization, err := GetAuth(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	info, err := s.store.UserPayments().Get(ctx, authorization.User.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	params := payments.AddPaymentMethodParams{
+		Token:      paymentMethodToken,
+		CustomerID: string(info.CustomerID),
+	}
+
+	method, err := s.pm.AddPaymentMethod(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+
+	projectPaymentInfo := ProjectPayment{
+		ProjectID:       projectID,
+		PayerID:         authorization.User.ID,
+		PaymentMethodID: method.ID,
+		CreatedAt:       time.Now(),
+		IsDefault:       isDefault,
+	}
+
+	create, err := s.store.ProjectPayments().Create(ctx, projectPaymentInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	return create, nil
+}
+
+// SetDefaultPaymentMethod set default payment method for given project
+func (s *Service) SetDefaultPaymentMethod(ctx context.Context, projectPaymentID uuid.UUID, projectID uuid.UUID) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	_, err = GetAuth(ctx)
+	if err != nil {
+		return err
+	}
+
+	projectPayment, err := s.store.ProjectPayments().GetDefaultByProjectID(ctx, projectID)
+	if err != nil {
+		return err
+	}
+	projectPayment.IsDefault = false
+
+	err = s.store.ProjectPayments().Update(ctx, *projectPayment)
+	if err != nil {
+		return err
+	}
+
+	projectPayment, err = s.store.ProjectPayments().GetByID(ctx, projectPaymentID)
+	if err != nil {
+		return err
+	}
+	projectPayment.IsDefault = true
+
+	return s.store.ProjectPayments().Update(ctx, *projectPayment)
+}
+
+// DeletePaymentMethod deletes selected payment method
+func (s *Service) DeletePaymentMethod(ctx context.Context, projectPayment uuid.UUID) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	_, err = GetAuth(ctx)
+	if err != nil {
+		return err
+	}
+
+	return s.store.ProjectPayments().Delete(ctx, projectPayment)
+}
+
+// GetProjectPaymentMethods retrieves project payment methods
+func (s *Service) GetProjectPaymentMethods(ctx context.Context, projectID uuid.UUID) ([]ProjectPayment, error) {
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	_, err = GetAuth(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	projectPaymentInfos, err := s.store.ProjectPayments().GetByProjectID(ctx, projectID)
+	if err != nil {
+		return nil, err
+	}
+
+	var projectPayments []ProjectPayment
+	for _, payment := range projectPaymentInfos {
+		pm, err := s.pm.GetPaymentMethod(ctx, payment.PaymentMethodID)
+		if err != nil {
+			return nil, err
+		}
+
+		cus, err := s.pm.GetCustomer(ctx, pm.CustomerID)
+		if err != nil {
+			return nil, err
+		}
+
+		user, err := s.store.Users().GetByEmail(ctx, cus.Email)
```

Comment:
that could fail

Reply:
this query returns only activated user

Reply:
Well, you were right. Found corner case when this is not a good approach